### PR TITLE
UNR-1484: Safety check for directory paths in SpatialOS Settings

### PR DESCRIPTION
**Pull request by chrisfields0**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
added code to check for invalid paths set in Editor Settings

#### Release note
REQUIRED: If SpatialOS GDK for Unreal - Editor Settings contains any directory paths that are invalid then they'll be reset to default paths.

#### Tests
Tested locally by editing .ini file to set fields to invalid path.

#### Primary reviewers
@m-samiec @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 1]
```diff
@@ -39,7 +43,51 @@ void USpatialGDKEditorSettings::PostInitProperties()
...
 	PlayInSettings->SaveConfig();
+
+	SafetyCheckSpatialOSDirectoryPaths();
+}
+
+void USpatialGDKEditorSettings::SafetyCheckSpatialOSDirectoryPaths()
+{
+	const FString path = FPaths::ConvertRelativePathToFull(FPaths::Combine(FPaths::ProjectDir(), TEXT("/../spatial/")));
+
+	FText text = FText::FromString(TEXT("The following directory paths are invalid in SpatialOS GDK for Unreal - Editor Settings:\n\n"));
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: Rename this variable to something more suitable, and perhaps keep as an FString until it's ready to be passed to the message dialog.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 2]
```diff
@@ -39,7 +43,51 @@ void USpatialGDKEditorSettings::PostInitProperties()
...
+
+	bool bFoundInvalidPath = false;
+
+	if (!FPaths::DirectoryExists(SpatialOSDirectory.Path))
+	{
+		SpatialOSDirectory.Path.Empty();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: nit: these clears are unnecessary

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 3]
```diff
@@ -1,6 +1,10 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 #include "SpatialGDKEditorSettings.h"
 #include "Settings/LevelEditorPlaySettings.h"
+#include "MessageDialog.h"
+
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp (hunk 1 above)
Comment (chrisfields0) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: Changed to FString

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp (hunk 3 above)
Comment (chrisfields0) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: removed the new line

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp (hunk 2 above)
Comment (chrisfields0) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: removed the empties

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 4]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 #include "SpatialGDKEditorSettings.h"
 #include "Settings/LevelEditorPlaySettings.h"
+#include "MessageDialog.h"
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: Can you fix these includes to be alphabetical?

Comment (cyberbibby) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: I recommend you guys use the .clang-format, no need to care about code formatting.

**Pull request merged**